File: Civitas_Ruby/lib/sorpresa_porCasaHotel.rb
module Civitas
  class Sorpresa_porCasaHotel < Sorpresa
    
    def initialize(valor, texto)
      super(texto)
      @valor = valor
    end
    
    def aplicarAJugador(actual, todos)
      if(jugadorCorrecto(actual, todos))
        informe(actual, todos)
        todos[actual].modificarSaldo(@valor * todos[actual].cantidadCasasHoteles())
      end
    end
    
    def informe(actual, todos)
      Diario.instance().ocurre_evento(todos[actual].to_s + " se ve afectado por una carta de tipo porCasaHotel y " + to_s())
    end
  end
end

File: Civitas_TUI/src/civitas/Sorpresa_porCasaHotel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;
import java.util.ArrayList;
/**
 *
 * @author francisco
 */
class Sorpresa_porCasaHotel extends Sorpresa{
    private int valor;
    
    Sorpresa_porCasaHotel(int valor1, String texto){
        super();
        this.valor = valor1;
        super.setTexto(texto);
    }
    
    @Override
    protected void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if (actual > -1 && actual < todos.size()){
            this.informe(actual, todos);
            todos.get(actual).modificarSaldo(valor*todos.get(actual).cantidadCasasHoteles());
        }
    }
    
    @Override
    protected void informe(int actual, ArrayList<Jugador> todos){
        if(super.jugadorCorrecto(actual, todos))
            Diario.getInstance().ocurreEvento(todos.get(actual).getNombre() + " se ve afectado por una sorpresa de tipo porCasaHotel " + super.toString());
    }
}
